### PR TITLE
Fix mcux SDK cache64 driver

### PR DESCRIPTION
**Pull request by mmahadevan108**
These changes have been submitted to the SDK team.

### CODE REVIEW

**Comment by mmahadevan108** [comment]
cc @DerekSnell

File: mcux/mcux-sdk/drivers/cache/cache64/fsl_cache.h [hunk 1]
```diff
@@ -79,12 +79,13 @@ uint32_t CACHE64_GetInstance(CACHE64_POLSEL_Type *base);
...
 /*!
- * brief Returns an instance number given physical memory address.
+ * brief Returns an instance numbers given physical memory address.
  *
  * param address The physical memory address.
+ * param pointer to store the index of the physical memory index
  * @return CACHE64_CTRL instance number starting from 0.
  */
-uint32_t CACHE64_GetInstanceByAddr(uint32_t address);
+uint32_t CACHE64_GetInstanceByAddr(uint32_t address, uint8_t *bufIndex);
```
Comment (decsny) [inline mcux/mcux-sdk/drivers/cache/cache64/fsl_cache.h:88]: why is the api changing?

Comment (mmahadevan108) [reply mcux/mcux-sdk/drivers/cache/cache64/fsl_cache.h:88]: This is an internal function and not an API change. This function was not functioning as intended. This function was returning a value which was used as an index for 2 arrays.

Comment (dleach02) [reply mcux/mcux-sdk/drivers/cache/cache64/fsl_cache.h:88]: if it is an internal function, then why is it defined in the header file? 

Is the SDK team okay with this change?

Comment (mmahadevan108) [reply mcux/mcux-sdk/drivers/cache/cache64/fsl_cache.h:88]: Working with the SDK team on this change.

**Comment by dleach02** [comment]
This needs to wait on the SDK 2.16.00 PR or these changes will be overwritten.

**Comment by decsny** [comment]
there is a merge conflict

**Review by dleach02 — CHANGES_REQUESTED** [review]
Need to add these patches to the patch list document

**Comment by mmahadevan108** [comment]
These patches have been added into the SDK repo and should be in our HAL when we do the next update. Should I still update the patches list?

**Comment by mmahadevan108** [comment]
@dleach02 PR updated with update to README. Added a patch that has been submitted to the SDK team after testing their new CACHE64 driver.

**Review by EmilioCBen — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]
Just to make sure I understand this change correctly, the CACHE64 invalidation only works correctly if a nonsecure address is written to `CSAR`?

**Comment by mmahadevan108** [comment]
Thats correct.

**Pull request merged**